lib/syskit/log/datastore/import.rb

```diff
@@ -324,26 +334,29 @@ def copy_roby_event_log(event_log_path, roby_sql_index)
             def copy_roby_event_log_one_cycle( # rubocop:disable Metrics/ParameterLists
                 in_reader, out_io, index_io,
                 rebuilder, roby_sql_index, in_stat,
-                metadata_update, digest
+                metadata_update
             )
                 pos = in_reader.tell
                 @reporter.current = pos
                 chunk = in_reader.read_one_chunk
                 cycle = in_reader.decode_one_chunk(chunk)
 
                 Roby::DRoby::Logfile.write_entry(out_io, chunk)
-                digest.update(chunk)
 
-                Roby::DRoby::Logfile::Index.write_one_cycle(index_io, pos, cycle)
+                if index_io
+                    Roby::DRoby::Logfile::Index.write_one_cycle(index_io, pos, cycle)
+                end
                 roby_sql_index.add_one_cycle(metadata_update, rebuilder, cycle)
                 true
             rescue Roby::DRoby::Logfile::TruncatedFileError => e
                 @reporter.warn e.message
                 @reporter.warn "truncating Roby log file"
-                index_io.rewind
-                Roby::DRoby::Logfile::Index.write_header(
-                    index_io, pos, in_stat.mtime
-                )
+                if index_io
+                    index_io&.rewind
```

Comment:
no need for the `&` here, index_io is already valid

Reply:
fixed